packages/tools/client-debugger/client-debugger-view/src/test/app/App.tsx

```diff
@@ -368,3 +387,22 @@ function CounterView(props: CounterViewProps): React.ReactElement {
 
 	return sharedCounter === undefined ? <Spinner /> : <CounterWidget counter={sharedCounter} />;
 }
+
+interface EmojiButtonViewProps {
+	emojiCellHandle: IFluidHandle<SharedCell<boolean>>;
+}
+
+function EmojiButtonView(props: EmojiButtonViewProps): React.ReactElement {
+	const { emojiCellHandle } = props;
+
+	const [emojiCell, setEmojiCell] = React.useState<SharedCell<boolean> | undefined>();
+
+	React.useEffect(() => {
+		emojiCellHandle.get().then(setEmojiCell, (error) => {
+			console.error(`Error encountered loading SharedCell: "${error}".`);
+			throw error;
+		});
+	}, [emojiCellHandle, setEmojiCell]);
```

Comment:
```suggestion
	}, [setEmojiCell]);
```
If props change the whole component re-renders and effects will run anyway, right?

Reply:
No, not necessarily. If the component is re-used across props changes (which React can do), our state will be incorrect. The general rule is that if your React hook depends on some subset of the input props, you have to specify them explicitly.

Reply:
React is much less functional than it appears 😕